libs/community/langchain_community/llms/cleanlab.py

```diff
@@ -0,0 +1,162 @@
+import logging
+from typing import Any, Dict, List, Mapping, Optional
+
+from langchain_core.callbacks import (
+    AsyncCallbackManagerForLLMRun,
+    CallbackManagerForLLMRun,
+)
+from langchain_core.language_models.llms import BaseLLM
+from langchain_core.outputs import Generation, LLMResult
+from langchain_core.utils import convert_to_secret_str, get_from_dict_or_env
+from pydantic import Field, PrivateAttr, SecretStr, model_validator
+
+from langchain_community.llms.utils import enforce_stop_tokens
+
+logger = logging.getLogger(__name__)
+
+
+class TrustworthyLanguageModel(BaseLLM):
+    """Cleanlab's Trustworthy Language Model (TLM).
+
+    To use, you should have the ``cleanlab-studio`` python package installed,
+    and the API key set either in the ``CLEANLAB_API_KEY`` environment variable,
+    or pass it as a named parameter to the constructor.
+    Sign up at app.cleanlab.ai to get a free API key.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_community.llms import TrustworthyLanguageModel
+            tlm = TrustworthyLanguageModel(
+                cleanlab_api_key="my_api_key",  # Optional if `CLEANLAB_API_KEY` is set
+                quality_preset="best"
+            )
+    """
+
+    _client: Any = PrivateAttr()  # :meta private:
+
+    cleanlab_api_key: Optional[SecretStr] = Field(default=None)
```

Comment:
you can remove some logic from the `validate_environment` function if you use `langchain_core.utils.secret_from_env`
```suggestion
    cleanlab_api_key: Optional[SecretStr] = Field(default_factory=secret_from_env("CLEANLAB_API_KEY"))
```